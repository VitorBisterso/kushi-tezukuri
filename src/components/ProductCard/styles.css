.productCard-container {
  display: flex;
  flex-direction: column;
  width: 220px;
  height: 320px;
  border-radius: var(--BORDER_RADIUS-BIG);
  box-shadow: 5px 10px 15px var(--COLOR-SHADOW);
}

.productCard-container > img {
  height: 150px;
  width: 100%;
  object-fit: cover;
  border-top-left-radius: var(--BORDER_RADIUS-BIG);
  border-top-right-radius: var(--BORDER_RADIUS-BIG);
  border-bottom: 1px solid var(--COLOR-SHADOW);
}

.productCard-container p {
  margin: 0;
}

.productCard-content {
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: 20px 10px;
  justify-content: space-between;
}

.productCard-title {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  font-family: 'Mali-SemiBold';
  color: var(--COLOR-LIGHT_BLACK);
}

.productCard-bottom {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-right: 10px;
}

.productCard-bottom > p {
  font-weight: bold;
  color: var(--COLOR-GOLD);
}

.productCard-units {
  display: flex;
  flex-direction: row;
  flex: 1;
  align-items: center;
  justify-content: center;
}

.productCard-units input {
  margin: 0 3px;
  width: 15px;
  height: 15px;
}
