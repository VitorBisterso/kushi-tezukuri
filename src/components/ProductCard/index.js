import React from 'react'
import PropTypes from 'prop-types'

import convertPriceToBrSyntax from '../../utils'

import cartIcon from '../../assets/img/cart.svg'
import editIcon from '../../assets/img/edit.svg'

import './styles.css'

const renderButton = isAdminPage => {
  const icon = isAdminPage ? editIcon : cartIcon
  const alt = isAdminPage ? 'editIcon' : 'cartIcon'

  return (
    <button type="button">
      <img src={icon} alt={alt} />
    </button>
  )
}

const ProductCard = ({ title, price, image, isAdminPage }) => (
  <div className="productCard-container">
    <img src={image} alt="product" />
    <div className="productCard-content">
      <p className="productCard-title">{title}</p>
      <div className="productCard-bottom">
        <p>{convertPriceToBrSyntax(price)}</p>
        <div className="productCard-units">
          <p>x</p>
          <input type="number" />
          <p>un.</p>
        </div>
        {renderButton(isAdminPage)}
      </div>
    </div>
  </div>
)

ProductCard.propTypes = {
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  image: PropTypes.node.isRequired,
  isAdminPage: PropTypes.bool.isRequired,
}

export default ProductCard
