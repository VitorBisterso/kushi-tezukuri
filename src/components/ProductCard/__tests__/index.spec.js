import React from 'react'
import { shallow } from 'enzyme'

import ProductCard from '../index'

import image from '../../../assets/img/mask.jpg'

describe('ProductCard', () => {
  it('should render correctly', () => {
    const component = shallow(
      <ProductCard image={image} title="Test card" price={5} isAdminPage />
    )

    expect(component).toMatchSnapshot()
  })
})
