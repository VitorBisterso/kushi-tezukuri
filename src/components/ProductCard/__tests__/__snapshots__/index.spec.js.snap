// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render correctly 1`] = `
<div
  className="productCard-container"
>
  <img
    alt="product"
    src=""
  />
  <div
    className="productCard-content"
  >
    <p
      className="productCard-title"
    >
      Test card
    </p>
    <div
      className="productCard-bottom"
    >
      <p>
        R$ 5,00
      </p>
      <div
        className="productCard-units"
      >
        <p>
          x
        </p>
        <input
          type="number"
        />
        <p>
          un.
        </p>
      </div>
      <button
        type="button"
      >
        <img
          alt="editIcon"
          src=""
        />
      </button>
    </div>
  </div>
</div>
`;
